Dialog docs (#2409)
microsoft/teams-ai: book/src/5.in-depth-guides/2.dialogs/1.creating-dialogs.md

# Creating Dialogs

> [!TIP]

> > If you're not familiar with how to build adaptive cards, check out [this guide](../1.cards/). Understanding their basics is a prerequisite for this guide.

Reviewer: FYI for accessibility, it's better to indicate as the link text what the link is

```suggestion
> If you're not familiar with how to build Adaptive Cards, check out the [cards guide](../1.cards/). Understanding their basics is a prerequisite for this guide.
```

Author: https://www.codecademy.com/resources/docs/markdown/links I can do it like this

Reviewer: That doc doesn't appear to talk about link naming, just titles.

https://www.w3.org/WAI/ARIA/apg/practices/names-and-descriptions/#composingeffectiveanduser-friendlyaccessiblenames

https://learn.microsoft.com/en-us/style-guide/accessibility/writing-all-abilities#write-brief-meaningful-and-focused-text

> Write brief but meaningful link text. Be descriptive—links should make sense without the surrounding text.

if you take out the sentence and keep just the link, 'this guide' doesn't tell the user where they're being taken to.

Author: good call! Updating to your suggestion

## Entry Point

> To open a dialog, you need to supply a special type of action as to the adaptive card. Once this button is clicked, the dialog will open and ask the application what to show.

Reviewer: Adaptive Card needs to be capitalized. 

```suggestion
To open a dialog, you need to supply a special type of action as to the Adaptive Card. Once this button is clicked, the dialog will open and ask the application what to show.
```

Reviewer: Could you do a pass in the PR?

```ts
{{#include ../../../generated-snippets/ts/index.snippet.dialog-entry-point.ts }}
```

## Handling Dialog Open Events

Once an action is executed to open a dialog, the Teams client will send an event to the agent to request what the content of the dialog should be. Here is how to handle this event:

```ts
{{#include ../../../generated-snippets/ts/index.snippet.dialog-open.ts }}
```

### Rendering A Card

You can render an adaptive card in a dialog by returning a card response.

```ts
{{#include ../../../generated-snippets/ts/index.snippet.dialog-simple-card.ts }}
```

> [!IMPORTANT]
> The action type for submitting a dialog must be `Action.Submit`. This is a requirement of the Teams client. If you use a different action type, the dialog will not be submitted and the agent will not receive the submission event.

### Rendering A Webpage

You can render a webpage in a dialog as well. There are some security requirements to be aware of here.

> 1. The webpage must be hosted on a domain that is allowlisted in the Teams app [manifest](../../4.teams/2.manifest.md) for the agent

Reviewer: ```suggestion
1. The webpage must be hosted on a domain that is allow-listed as `validDomains` in the Teams app [manifest](../../4.teams/2.manifest.md) for the agent
```
